lib/Cro/Uri.pm6

```diff
@@ -250,24 +250,21 @@ class Cro::Uri does Cro::ResourceIdentifier {
 
         method path-abempty($/) {
             my $result = '';
-            for @$<segment> {
-                $result ~= '/';
-                $result ~= $_.ast with $_<pchars>;
-            }
+            $result ~= '/' ~ ($_<pchars> ?? $_<pchars>.ast !! '') for @$<segment>;
             make $result;
         }
 
         method path-absolute($/) {
             my $result = '/';
             $result ~= $_<pchars>.ast with $<segment-nz>;
-            $result ~= '/' ~ $_<pchars>.ast for @$<segment>;
+            $result ~= '/' ~ ($_<pchars> ?? $_<pchars>.ast !! ~$_) for @$<segment>;
```

Comment:
I don't have a firm understanding of all the regexy bits going on.
Before the change it errored out if there was no `pchars`. Now it's to be decided if in that case the non-pchars text should be discarded or kept. Irrespective of what we do, we'll change the semantics. My bet was that we want to keep it. What do you think?

Reply:
I guess we need to be clear with what original path it creates the errors, what the regex / token is that created the `Match` object, and what the result needs to be.

All of that is pretty unclear to me  :-(

Reply:
segment is literally only

```
token segment {
    <pchars>?
}
```

So it doesn't make a difference. What do you think would be cleaner? `~$_` or `''`?

Reply:
In that case, I'd say: `$result ~= '/' ~ (~$_ if $<pchars>) for @$<segment>;` ?

since there's nothing else in the `token`, we can safely stringify the whole `Match` **if** `<pchars>` matched.

Reply:
What does `$<pchars>.ast` return? If it's the thing `method pchars()` `make`d, then `~$_` is different from `$<pchars>.ast`.

Reply:
Ah, duh  :-)  Forgot about the `.ast` part.  `$result ~= '/' ~ (.ast with $<pchars>) for @$<segment>;` then?

Reply:
There you go.